lucene/facet/src/java/org/apache/lucene/facet/DrillSideways.java

```diff
@@ -390,18 +398,44 @@ protected boolean scoreSubDocsAtOnce() {
     return false;
   }
 
-  /** Result of a drill sideways search, including the {@link Facets} and {@link TopDocs}. */
+  /**
+   * Result of a drill sideways search, including the {@link Facets} and {@link TopDocs} {@link
+   * FacetsCollector} for drill down and drill sideways passed to {@link #buildFacetsResult} to
+   * compute {@link Facets} are also exposed here as an alternative to subclassing DrillSideways to
+   * obtain them
+   */
   public static class DrillSidewaysResult {
     /** Combined drill down and sideways results. */
     public final Facets facets;
 
     /** Hits. */
     public final TopDocs hits;
 
+    /** FacetsCollector for drill down. */
```

Comment:
I might make this documentation a little more verbose since this is a tricky domain and really only exposed for less-traditional use-cases. Maybe something like:

```suggestion
    /** 
    * FacetsCollector populated based on hits that match the full DrillDownQuery,
    * treating all drill down dimensions as required clauses. Useful for advanced 
    * use-cases that want to compute Facets results separate from the provided 
    * Facets in this result.
    */
```